warden/x/warden/keeper/keychain.go

```diff
@@ -2,32 +2,35 @@ package keeper
 
 import (
 	"context"
-	"fmt"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	types "github.com/warden-protocol/wardenprotocol/warden/x/warden/types/v1beta3"
 )
 
-func (k Keeper) deductKeychainFees(
-	ctx context.Context,
-	maxKeychainFee sdk.Coins,
-	keychainAccAddress sdk.AccAddress,
-	keychainFees sdk.Coins,
-	creator sdk.AccAddress) error {
-	if maxKeychainFee.Empty() || keychainFees.Empty() {
-		return fmt.Errorf("fees cannot be empty")
+// DeductKeychainFees is to deduct fees from creator's address to warden escrow address
+func (k Keeper) DeductKeychainFees(ctx context.Context, creator sdk.AccAddress, fees sdk.Coins) error {
```

Comment:
let's keep these private so we are sure they can only be used from inside this keeper (ensuring the tokens are managed properly)

```suggestion
func (k Keeper) deductKeychainFees(ctx context.Context, creator sdk.AccAddress, fees sdk.Coins) error {
```